### PR TITLE
feat: add preliminary alphafold support

### PR DESCRIPTION
**Pull request by jherr-dev**
Adds basic support for running AlphaFold via Toolchest. Code needs to be cleaned up and better documented. Currently limited to 1 input fasta.

`use_reduced_dbs` and `is_prokaryote_list` are currently disabled until further implementation and testing is done. Integration will come with reduced dbs since full dbs take 45 minutes to an hour to run even on simple input.

### CODE REVIEW

File: toolchest_client/tools/alphafold.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"""
+toolchest_client.tools.bowtie2
```
Comment (lebovic) [inline toolchest_client/tools/alphafold.py:2]: sp: bowtie2 -> alphafold

File: toolchest_client/tools/alphafold.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is the bowtie2 implementation of the Tool class.
```
Comment (lebovic) [inline toolchest_client/tools/alphafold.py:5]: sp: bowtie2 -> alphafold

File: toolchest_client/tools/alphafold.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+"""
+from toolchest_client.files import OutputType
+from datetime import date
+from . import Tool
+
+
+class AlphaFold(Tool):
+    """
+    The bowtie2 implementation of the Tool class.
```
Comment (lebovic) [inline toolchest_client/tools/alphafold.py:14]: sp: bowtie2 -> alphafold

File: toolchest_client/tools/alphafold.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    """
+    def __init__(self, inputs, output_path, model_preset=None, max_template_date=None, use_reduced_dbs=False,
+                 is_prokaryote_list=None):
+        tool_args = (
+            (f"--model_preset={model_preset} " if model_preset is not None else "") +
+            (f"--max_template_date={max_template_date} " if max_template_date is not None
+             else f"--max_template_date={date.today().strftime('%Y-%m-%d')} ")  # +
+            # (f"--is_prokaryote_list={is_prokaryote_list.join(',')} " if is_prokaryote_list is not None else "") +
+            # ("--db_preset=reduced_dbs " if use_reduced_dbs else "")
+        )
```
Comment (lebovic) [inline toolchest_client/tools/alphafold.py:24]: It looks like most tool_arg processing is happening in `api.py` for other tools.

Was this an opinionated choice? I don't have a preference.

File: toolchest_client/tools/alphafold.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            output_name='output.tar.gz',
+            inputs=inputs,
+            min_inputs=1,
+            max_inputs=1,  # May need to be increased in future
+            database_name="alphafold_standard",
+            database_version="2.1.2",
+            parallel_enabled=False,
+            output_type=OutputType.GZ_TAR,
+            output_path=output_path,
+            output_is_directory=True,
```
Comment (lebovic) [inline toolchest_client/tools/alphafold.py:38]: Missing passing of `**kwargs`

**Review by lebovic — COMMENTED** [review]
A few small changes to make, otherwise looks good – I'll make them now for the sake of time

**Comment by lebovic** [comment]
@bcai2 @jherr-dev merging to bring this into staging, but do still want to finish out these convos

File: toolchest_client/tools/alphafold.py (hunk 4 above)
Comment (jherr-dev) [reply toolchest_client/tools/alphafold.py:24]: No I just started implementing these before adding AlphaFold to `api.py` I'll move it to `api.py` to match convention.

**Pull request merged**